src/components/product/product.service.ts

```diff
@@ -23,106 +37,141 @@ import {
 
-      if (input.produces) {
-        await this.db
-          .query()
-          .match([
-            [node('product', 'Product', { id, active: true })],
-            [node('pr', 'Producible', { id: input.produces, active: true })],
-          ])
-          .create([
-            node('product'),
-            relation('out', '', 'produces', {
-              active: true,
-              createdAt: DateTime.local(),
-            }),
-            node('pr'),
-          ])
-          .run();
-      }
-    } catch (e) {
-      this.logger.warning('Failed to create product', {
-        exception: e,
-      });
+    const result = await query.first();
 
-      throw new ServerException('Failed to create product');
+    if (!result) {
+      throw new ServerException('failed to create default product');
     }
 
+    const id = result.id;
+
+    // add root admin to new product as an admin
+    await this.db.addRootAdminToBaseNodeAsAdmin(id, 'Product');
+
+    this.logger.debug(`product created, id ${id}`);
+
     return this.readOne(id, session);
+    // try {
+    //   await this.db.createNode({
+    //     session,
+    //     type: Product,
+    //     input: {
+    //       id,
+    //       ...input,
+    //       ...(input.methodology
+    //         ? { approach: MethodologyToApproach[input.methodology] }
+    //         : {}),
+    //     },
+    //     acls,
+    //   });
+
+    //   if (input.produces) {
+    //     await this.db
+    //       .query()
+    //       .match([
+    //         [node('product', 'Product', { id, active: true })],
+    //         [node('pr', 'Producible', { id: input.produces, active: true })],
+    //       ])
+    //       .create([
+    //         node('product'),
+    //         relation('out', '', 'produces', {
+    //           active: true,
+    //           createdAt: DateTime.local(),
+    //         }),
+    //         node('pr'),
+    //       ])
+    //       .run();
+    //   }
+    // } catch (e) {
+    //   this.logger.warning('Failed to create product', {
+    //     exception: e,
+    //   });
+
+    //   throw new ServerException('Failed to create product');
+    // }
+
+    // return this.readOne(id, session);
   }
 
   async readOne(id: string, session: ISession): Promise<AnyProduct> {
-    const result = await this.db.readProperties({
-      session,
-      id,
-      nodevar: 'product',
-      props: [
-        'id',
-        'createdAt',
-        'type',
-        'books',
-        'mediums',
-        'purposes',
-        'approach',
-        'methodology',
-      ],
-    });
+    const props = ['mediums', 'purposes', 'methodology'];
+    const query = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .call(matchUserPermissions, 'Product', id)
+      .call(addAllSecureProperties, ...props)
+      .with([
+        ...props.map(addPropertyCoalesceWithClause),
+        'coalesce(node.id) as id',
+        'coalesce(node.createdAt) as createdAt',
```

Comment:
FYI there's no need to `coalesce` if there's not more than one argument in the function call.